plumbing/format/packfile/scanner.go

```diff
@@ -56,19 +55,27 @@ type Scanner struct {
 // NewScanner returns a new Scanner based on a reader, if the given reader
 // implements io.ReadSeeker the Scanner will be also Seekable
 func NewScanner(r io.Reader) *Scanner {
-	seeker, ok := r.(io.ReadSeeker)
-	if !ok {
-		seeker = &trackableReader{Reader: r}
-	}
+	_, ok := r.(io.ReadSeeker)
 
 	crc := crc32.NewIEEE()
 	return &Scanner{
-		r:          newTeeReader(newByteReadSeeker(seeker), crc),
+		r:          newScannerReader(r, crc),
 		crc:        crc,
 		IsSeekable: ok,
 	}
 }
 
+func (s *Scanner) Reset(r io.Reader) {
```

Comment:
It would be nice to hace a test that reuses scanner with `Reset`. Other than that LGTM! Thanks!

Reply:
Good shout! I noticed a problem whilst figuring out how/what to test. Even when the reader is seekable, I'm returning the `r.offset` when `whence == io.SeekCurrent && offset == 0`, which incorrect assumes that the reader provided to start with was set at the beginning of the file.

I'll fix that and add a test shortly!

Reply:
I've added two tests. `TestReaderReset` checks basics usage and that fields are correctly reset. `TestReaderResetSeeks` checks that the seek behavior is correct.